app/domain/authentication/authn_jwt/signing_key/create_signing_key_factory.rb

```diff
@@ -0,0 +1,82 @@
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # Factory that returns the interface implementation of FetchSigningKey
+      CreateSigningKeyFactory ||= CommandClass.new(
+        dependencies: {
+          fetch_provider_uri_signing_key_class: Authentication::AuthnJwt::SigningKey::FetchProviderUriSigningKey,
+          fetch_jwks_uri_signing_key_class: Authentication::AuthnJwt::SigningKey::FetchJwksUriSigningKey,
+          logger: Rails.logger
+        },
+        inputs: %i[authentication_parameters]
+      ) do
+
+        def call
+          @logger.debug(LogMessages::Authentication::AuthnJwt::SelectingSigningKeyInterface.new)
+          validate_key_configuration
+          create_signing_key
+        end
+
+        private
+
+        def create_signing_key
+          if provider_uri_has_valid_configuration?
+            @logger.info(
+              LogMessages::Authentication::AuthnJwt::SelectedSigningKeyInterface.new(PROVIDER_URI_INTERFACE_NAME)
+            )
+            fetch_provider_uri_signing_key
+          elsif jwks_uri_has_valid_configuration?
+            @logger.info(
+              LogMessages::Authentication::AuthnJwt::SelectedSigningKeyInterface.new(JWKS_URI_INTERFACE_NAME)
+            )
+            fetch_jwks_uri_signing_key
+          end
+        end
+
+        def validate_key_configuration
```

Comment:
please move this upwards to be implemented according to the call order

Reply:
Done